src/color.rs

```diff
@@ -524,13 +436,114 @@ fn parse_color_function(name: &str, arguments: &mut Parser) -> Result<Color, ()>
         255
     };
 
-    /*
-    let alpha = if has_alpha {
-        try!(arguments.expect_comma());
-        clamp_f32(try!(arguments.expect_number()))
-    } else {
-        255
-    };*/
     try!(arguments.expect_exhausted());
     rgba(red, green, blue, alpha)
 }
+
+
+#[inline]
+fn parse_rgb_components_rgb(arguments: &mut Parser) -> Result<(u8, u8, u8, bool), ()> {
+    let red: u8;
+    let green: u8;
+    let blue: u8;
+    let mut uses_commas = false;
+
+    // Either integers or percentages, but all the same type.
+    // https://drafts.csswg.org/css-color/#rgb-functions
+    match try!(arguments.next()) {
+        Token::Number(NumericValue { value: v, .. }) => {
+            red = clamp_i32(v as i32);
+            green = clamp_i32(match try!(arguments.next()) {
+                Token::Number(NumericValue { value: v, .. }) => v,
+                Token::Comma => {
+                    uses_commas = true;
+                    try!(arguments.expect_number())
+                }
+                _ => return Err(())
+            } as i32);
+            if uses_commas {
+                try!(arguments.expect_comma());
+            }
+            blue = clamp_i32(try!(arguments.expect_number()) as i32);
+        }
+        Token::Percentage(ref v) => {
+            red = clamp_f32(v.unit_value);
+            green = clamp_f32(match try!(arguments.next()) {
+                Token::Percentage(ref v) => v.unit_value,
+                Token::Comma => {
+                    uses_commas = true;
+                    try!(arguments.expect_percentage())
+                }
+                _ => return Err(())
+            });
+            if uses_commas {
+                try!(arguments.expect_comma());
+            }
+            blue = clamp_f32(try!(arguments.expect_percentage()));
+        }
+        _ => return Err(())
+    };
+    return Ok((red, green, blue, uses_commas));
+}
+
+#[inline]
+fn parse_rgb_components_hsl(arguments: &mut Parser) -> Result<(u8, u8, u8, bool), ()> {
+    let mut uses_commas = false;
+    // Hue given as an angle
+    // https://drafts.csswg.org/css-values/#angles
+    let hue_degrees = match try!(arguments.next()) {
+        Token::Number(NumericValue { value: v, .. }) => v,
+        Token::Dimension(NumericValue { value: v, .. }, unit) => {
+            match_ignore_ascii_case! { &*unit,
+                "deg" => v,
+                "grad" => v * 360. / 400.,
+                "rad" => v * 360. / (2. * PI as f32),
```

Comment:
One last nit: Let's use `std::f32::consts::PI` instead of the `f64` version unless there's a good reason for that.